File: 30DaysOfCode/Day 19 - Interfaces.java
import java.io.*;
import java.util.*;

interface AdvancedArithmetic{
   int divisorSum(int n);
}

//To DO
class Calculator implements AdvancedArithmetic{
	int sum=0;
	
	@Override
	public int divisorSum(int n) {
		for(int i=1;i<=n;i++){
			if(n%i == 0){
				sum += i;
			}
		}
		return sum;
	}	
}

class Solution {

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        int n = scan.nextInt();
        scan.close();
        
      	AdvancedArithmetic myCalculator = new Calculator(); 
        int sum = myCalculator.divisorSum(n);
        System.out.println("I implemented: " + myCalculator.getClass().getInterfaces()[0].getName() );
        System.out.println(sum);
    }
}


Sample Input:

6

Sample Output:

I implemented: AdvancedArithmetic
12

File: 30DaysOfCode - Kotlin/Day_19_Interfaces.kt
import java.util.*

internal interface AdvancedArithmetic {
    fun divisorSum(n: Int): Int
}

internal class Calculator1 : AdvancedArithmetic {
    override fun divisorSum(n: Int): Int {
        var sum = 0
        for (i in 1..n) {
            if (n % i == 0) {
                sum = sum + i
            }
        }
        return sum
    }
}

fun main(args: Array<String>) {

    val scan  = Scanner(System.`in`)
    val n = scan.nextInt()
    scan.close()

    val myCalculator: AdvancedArithmetic = Calculator1()
    val sum = myCalculator.divisorSum(n)
    println("I implemented: " + myCalculator.javaClass.getInterfaces()[0].getName())
    println(sum)
}
